### PR TITLE
add usage instructions for tyk-sync examples and udg-examples page (TT-8179)

### PR DESCRIPTION
**Pull request by pvormste**
## Related Issue (if applicable)
https://tyktech.atlassian.net/browse/TT-8179

## Description
Add usage instructions for `tyk-sync examples` and a proper page for `UDG Examples`

## Motivation and Context
The usage of `tyk-sync` examples is currently undocumented and therefore hidden. This PR will change it.

## Preview Link
- https://deploy-preview-2417--tyk-docs.netlify.app/docs/nightly/tyk-sync/
- https://deploy-preview-2417--tyk-docs.netlify.app/docs/nightly/universal-data-graph/udg-examples/

## Types of changes

- [ ] Fixing typo (please merge to production)
- [x] Documenting a new feature (please merge to production)
- [ ] Documentation for future release (please do not merge to production)
- [ ] Something else (please add if needs merging to production or not)

## Checklist

- [x] I have [reviewed the guidelines](../CONTRIBUTING.md) for contributing to this repository.
- [x] I have [read the technical guidelines](../CONTRIBUTING-TECHNICAL-GUIDE.md) for contributing to this repository.
- [x] Make sure you have started *your change* off *our latest `master`*.
- [x] I have added a preview link.

### CODE REVIEW

**Review by tbuchaillot — APPROVED** [review]
LGTM 🚀

**Review by jay-deshmukh — APPROVED** [review]

File: tyk-docs/content/tyk-sync.md [hunk 1]
```diff
@@ -178,6 +180,50 @@ Flags:
     --apis               Specific api_id's selection (optional)
 ```
 
+### Examples Command
+
+The examples command lists all examples from our official [Tyk examples](https://github.com/TykTechnologies/tyk-examples) repository.
+```
```
Comment (letzya) [inline tyk-docs/content/tyk-sync.md]:
```suggestion
```bash
```

File: tyk-docs/content/tyk-sync.md [hunk 2]
```diff
@@ -178,6 +180,50 @@ Flags:
...
+  publish     Publish a specific example to a gateway or dashboard by using its location
+  show        Shows details of a specific example by using its location
+
+Flags:
+  -h, --help   help for examples
+```
+
+### Examples Show Command
+Shows more details about a specific example by using its location.
+```
```
Comment (letzya) [inline tyk-docs/content/tyk-sync.md]:
```suggestion
```bash
```

File: tyk-docs/content/tyk-sync.md [hunk 3]
```diff
@@ -282,3 +328,47 @@ To check the current Tyk Sync version, we need to run the version command:
 tyk-sync version
 v1.2
 ```
+
+## Example: Import Tyk example into Dashboard
+
+To list all available examples you need to run this command:
+```
```
Comment (letzya) [inline tyk-docs/content/tyk-sync.md]:
```suggestion
```bash
```

File: tyk-docs/content/tyk-sync.md [hunk 4]
```diff
@@ -178,6 +180,50 @@ Flags:
...
+  show        Shows details of a specific example by using its location
+
+Flags:
+  -h, --help   help for examples
+```
+
+### Examples Show Command
+Shows more details about a specific example by using its location.
+```
+Usage:
```
Comment (letzya) [inline tyk-docs/content/tyk-sync.md:202]: Not sure I understand why we keep showing the help output. can we have the output of the show subcommand?

File: tyk-docs/content/tyk-sync.md [hunk 5]
```diff
@@ -178,6 +180,50 @@ Flags:
...
+  tyk-sync examples show [flags]
+
+Flags:
+  -h, --help              help for show
+  -l, --location string   Location to example
+```
+
+### Examples Publish Command
+Publishs an example by using its location.
+```
```
Comment (letzya) [inline tyk-docs/content/tyk-sync.md]:
```suggestion
```bash
```

File: tyk-docs/content/tyk-sync.md [hunk 6]
```diff
@@ -178,6 +180,50 @@ Flags:
...
+Usage:
+  tyk-sync examples publish [flags]
```
Comment (letzya) [inline tyk-docs/content/tyk-sync.md:214]: again, we don't need the help output, just the publish output

File: tyk-docs/content/tyk-sync.md [hunk 7]
```diff
@@ -282,3 +328,47 @@ To check the current Tyk Sync version, we need to run the version command:
...
+To list all available examples you need to run this command:
+```
+tyk-sync examples
+LOCATION           NAME                               DESCRIPTION
+udg/vat-checker    VAT number checker UDG             Simple REST API wrapped in GQL using Universal Data Graph that allows user to check validity of a VAT number and display some details about it.
+udg/geo-info       Geo information about the World    Countries GQL API extended with information from Restcountries
+```
+
+It's also possible to show more details about an example by using its location:
+```
```
Comment (letzya) [inline tyk-docs/content/tyk-sync.md]:
```suggestion
```bash
```

File: tyk-docs/content/tyk-sync.md [hunk 8]
```diff
@@ -282,3 +328,47 @@ To check the current Tyk Sync version, we need to run the version command:
...
+
+To list all available examples you need to run this command:
+```
+tyk-sync examples
+LOCATION           NAME                               DESCRIPTION
+udg/vat-checker    VAT number checker UDG             Simple REST API wrapped in GQL using Universal Data Graph that allows user to check validity of a VAT number and display some details about it.
+udg/geo-info       Geo information about the World    Countries GQL API extended with information from Restcountries
+```
+
+It's also possible to show more details about an example by using its location:
```
Comment (letzya) [inline tyk-docs/content/tyk-sync.md]:
```suggestion
It's also possible to show more details about an example by using its location. For example, based on the output from `tyk-sync examples` above, we can use the location of the example "VAT number checker UDG" to get more information:
```

File: tyk-docs/content/tyk-sync.md [hunk 9]
```diff
@@ -282,3 +328,47 @@ To check the current Tyk Sync version, we need to run the version command:
...
+
+FEATURES
+- REST Datasource
+
+MIN TYK VERSION
+5.0
+```
+
+To publish it into the Dashboard you will need to use this command:
+```
```
Comment (letzya) [inline tyk-docs/content/tyk-sync.md]:
```suggestion
```bash
```

**Review by letzya — CHANGES_REQUESTED** [review]
please update per comments

**Review by agata-wit — APPROVED** [review]

**Comment by buger** [comment]
Maybe also include references from some UDG getting started page to this doc?

**Comment by agata-wit** [comment]
> Maybe also include references from some UDG getting started page to this doc?

I don't think it's necessary for a couple of reasons:

- this page will be right under "Getting started" section in the navigation menu, it will really be the easiest thing ever to find it
- there's already quite a few links in here, adding more to content that can be easily seen in navigation menu seems redundant and will make the page more cluttered

**Comment by letzya** [comment]
@agata-wit I agree with @buger - we need more links - it's not about humans finding it and if it's easy for them, but about crawlers and SEO. Can you please add links whenever possible?

**Comment by agata-wit** [comment]
ok, we could also add some links on the main page: https://deploy-preview-2417--tyk-docs.netlify.app/docs/nightly/universal-data-graph/
I already expanded it with a few links like this: https://tyk.io/docs/universal-data-graph/  (at the bottom)
we could add one more there and be like: Try an example

I'll do it in a separate PR since this is merged already.

**Pull request merged**